launcher_scripts/nemo_launcher/core/stages.py

```diff
@@ -137,12 +136,31 @@ def save_stage_hydra_config(
             temp_config["data_config"] = stage_cfg.run.name
             stage_cfg = OmegaConf.create(temp_config)
 
-        _hydra_interpolation(stage_cfg)
+        self.hydra_interpolation(stage_cfg)
 
         cfg_save_path = job_path.config_file
         omegaconf.OmegaConf.save(stage_cfg, cfg_save_path)
         return cfg_save_path
 
+    def hydra_interpolation(self, cfg: OmegaConf) -> None:
```

Comment:
So I looked into your test failures and it looks like it's because you change the method: `save_stage_hydra_config` from a static method to an instance method. Other stages outside of stages.py expected it to be a staticmethod. 

I would encourage you to revert the staticmethod and change this method to a static method, since it creates a larger code change than necesary. Unless I'm missing something, there's nothing that requires the state of the object in `hydra_interpolation`, so it seems like it should be a static method.